py-polars/tests/unit/interchange/test_roundtrip.py

```diff
@@ -109,3 +241,13 @@ def test_roundtrip_pyarrow_boolean_midbyte_slice() -> None:
     result: pl.DataFrame = pl.from_arrow(df_pa)  # type: ignore[assignment]
 
     assert_frame_equal(result, df)
+
+
+@pytest.mark.xfail(
+    reason="Bug in pandas: https://github.com/pandas-dev/pandas/issues/56712"
+)
+def test_from_dataframe_pandas_timestamp_ns() -> None:
+    df = pl.Series("a", [datetime(2000, 1, 1)], dtype=pl.Datetime("ns")).to_frame()
+    df_pd = df.to_pandas(use_pyarrow_extension_array=True)
+    result = pl.from_dataframe(df_pd)
+    assert_frame_equal(result, df)
```

Comment:
this is fixed on `main`, and 2.2 should be out soon (this month) - simplest thing might be to just wait until it's out? else `xfail` based on the pandas version?

Reply:
I made it xfail based on the pandas version 👍